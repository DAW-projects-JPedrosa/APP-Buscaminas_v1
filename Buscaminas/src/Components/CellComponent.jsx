import React from 'react';
import { useState } from 'react'

function CellComponent({cell, onClick, onContextMenu}) {
  const [internalCell, setCell] = useState(cell);

  function handleClick(){
    setCell(onClick())
  }

  function handleRightClick(e){
    setCell(onContextMenu(e))
  }

  return <>
        <div
        className={`cell ${internalCell.defused ? 'defused' : ''} ${internalCell.flagged ? 'flagged' : ''} ${internalCell.exploded ? 'exploded' : ''}`}
        id={`${internalCell.row}-${internalCell.column}`}
        onClick={handleClick}
        onContextMenu={handleRightClick}
        >
          {internalCell.defused && internalCell.surroundingMinesCount > 0 && internalCell.surroundingMinesCount}
          {internalCell.flagged && 'ðŸš©'}
          {internalCell.exploded && 'ðŸ’£'}
        </div>
    </>
};

export default CellComponent;
