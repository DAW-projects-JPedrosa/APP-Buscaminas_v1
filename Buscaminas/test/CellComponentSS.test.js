import React from 'react';
import Renderer from "react-test-renderer";
import CellComponent from '../src/Components/CellComponent';

it("Renders a cell without any state", () => {
    //Arrange
    const cell = {row:0, column:0, flagged:false, exploded:false, defused:false, surroundingMinesCount: 0};

    //Act
    const cellComponent = Renderer.create(
        <CellComponent cell={cell} onClick={()=>{}} onContextMenu={()=>{}}></CellComponent>
    )

    //Assert
    let tree = cellComponent.toJSON();
    expect(tree).toMatchSnapshot();
});

it("Renders a cell with defused state and 0 mines around", () => {
    //Arrange
    const cell = {row:1, column:2, flagged:false, exploded:false, defused:true, surroundingMinesCount: 0};

    //Act
    const cellComponent = Renderer.create(
        <CellComponent cell={cell} onClick={()=>{}} onContextMenu={()=>{}}></CellComponent>
    )

    //Assert
    let tree = cellComponent.toJSON();
    expect(tree).toMatchSnapshot();
})

it("Renders a cell with defused state and 3 mines around", () => {
    //Arrange
    const cell = {row:1, column:2, flagged:false, exploded:false, defused:true, surroundingMinesCount: 3};

    //Act
    const cellComponent = Renderer.create(
        <CellComponent cell={cell} onClick={()=>{}} onContextMenu={()=>{}}></CellComponent>
    )

    //Assert
    let tree = cellComponent.toJSON();
    expect(tree).toMatchSnapshot();
})

it("Renders a cell with flagged state", () => {
    //Arrange
    const cell = {row:1, column:3, flagged:true, exploded:false, defused:false, surroundingMinesCount: 3};

    //Act
    const cellComponent = Renderer.create(
        <CellComponent cell={cell} onClick={()=>{}} onContextMenu={()=>{}}></CellComponent>
    )

    //Assert
    let tree = cellComponent.toJSON();
    expect(tree).toMatchSnapshot();
})

it("Renders a cell with exploded state", () => {
    //Arrange
    const cell = {row:1, column:3, flagged:false, exploded:true, defused:false, surroundingMinesCount: 3};

    //Act
    const cellComponent = Renderer.create(
        <CellComponent cell={cell} onClick={()=>{}} onContextMenu={()=>{}}></CellComponent>
    )

    //Assert
    let tree = cellComponent.toJSON();
    expect(tree).toMatchSnapshot();
})