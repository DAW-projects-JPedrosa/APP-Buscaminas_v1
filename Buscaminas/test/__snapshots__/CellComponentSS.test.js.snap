// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders a cell with defused state and 0 mines around 1`] = `
<div
  className="cell defused  "
  id="1-2"
  onClick={[Function]}
  onContextMenu={[Function]}
/>
`;

exports[`Renders a cell with defused state and 3 mines around 1`] = `
<div
  className="cell defused  "
  id="1-2"
  onClick={[Function]}
  onContextMenu={[Function]}
>
  3
</div>
`;

exports[`Renders a cell with exploded state 1`] = `
<div
  className="cell   exploded"
  id="1-3"
  onClick={[Function]}
  onContextMenu={[Function]}
>
  ðŸ’£
</div>
`;

exports[`Renders a cell with flagged state 1`] = `
<div
  className="cell  flagged "
  id="1-3"
  onClick={[Function]}
  onContextMenu={[Function]}
>
  ðŸš©
</div>
`;

exports[`Renders a cell without any state 1`] = `
<div
  className="cell   "
  id="0-0"
  onClick={[Function]}
  onContextMenu={[Function]}
/>
`;
